### PR TITLE
fix: `getValidated()` when validation multiple asterisk

### LINKED ISSUE #9219: Bug: getValidated() missing data when double wildcard rule
### PHP Version

8.3

### CodeIgniter4 Version

4.5.5

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

macOS, Linux

### Which server did you use?

apache

### Database

MySQL 5.7, 10.5.22-MariaDB

### What happened?

When running validation in the controller and grabbing the validated data with `$this->validator->getValidated()`, the data for double wild card rules (eg `dates.*.*`) is missing.

### Steps to Reproduce

Implement this controller (or the logic):
```php
<?php

namespace App\Controllers\Back;

use App\Controllers\BaseController;

class TestController extends BaseController
{
    public function test()
    {
        $data = [
            'id'    => 1,
            'dates' => [
                23 => [
                    45 => '3.4.24',
                ],
            ],
        ];

        $this->validateData($data, [
            'id'        => 'required',
            'dates.*.*' => 'required',
        ]);

        $validated = $this->validator->getValidated();

        d($data, $validated);
    }
}
```

### Expected Output

The validated data including in this case the date array.

### Anything else?

I think the problem is that the validation rule ends in a double wildcard, because when i change the array and rules to this it works:
```php
$data = [
    'id'    => 1,
    'dates' => [
        23 => [
            45 => [
                'date' => '3.4.24',
            ],
        ],
    ],
];

$this->validateData($data, [
    'id'             => 'required',
    'dates.*.*.date' => 'required',
]);

$validated = $this->validator->getValidated();

d($data, $validated);
```
Does CI4 not like rules ending with wildcards?

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Fixed #9219

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.5.6.rst [hunk 1]
```diff
@@ -30,6 +30,8 @@ Deprecations
 Bugs Fixed
 **********
 
+- **Validation:** Fixed `getValidated()` when validation multiple asterisk.
```
Comment (michalsn) [inline user_guide_src/source/changelogs/v4.5.6.rst:33]:
```suggestion
- **Validation:** Fixed the `getValidated()` method that did not return valid data when validation rules used multiple asterisks.
```

**Review by michalsn — COMMENTED** [review]
Do we have any tests that cover multiple asterisks, when validation rules failed?

**Review by michalsn — APPROVED** [review]
Thank you!

**Comment by michalsn** [comment]
@ddevsr Can you rebase?

**Review by michalsn — APPROVED** [review]
Thank you.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**